### ISSUE TITLE
Image artifacts with certain gifs

### ISSUE THREAD

**Issue by shanecbauman** [hzeller/rpi-rgb-led-matrix#1152]
First off, thanks for the amazing library!  It is so fun to work with.  Anyways, here is the problem im experiencing:

Currently im having an issue with artifacts showing up on only certain GIFs that I play.  Some GIFs run flawlessly,  while others have weird flickering artifacts that repeat consistently on each loop through the GIF.  

The artifacting can be as little as a small flicker at one point in the GIF, to the whole GIF flashing in and out while it is playing.

The hardware im using is:
rpi3 (running dietpi)
adafruit hat with jumper wire mod
5v 20a PSU
2x LED panel: P4-256*128-2121-A5 (its a 32x64 panel from aliexpress)

Im launching the led-image-viewer with the following command:
sudo ./utils/led-image-viewer --led-gpio-mapping=adafruit-hat-pwm --led-chain=2 --led-pixel-mapper="U-mapper;Rotate:180" --led-cols=64 --led-rows=32

The settings i have played with so far:
--led-pwm-bits
--led-limit-refresh
--led-pwm-lsb-nanoseconds
--led-pwm-dither-bits

Troubleshooting steps I have taken:
-added "dtparam=audio=off" to /boot/config.txt
-added "isolcpus=3" to /boot/config.txt
-set NTP to only run on boot
-sudo systemctl stop cron

The thing that confuses me is that some GIFs run flawlessly, with no artifacting at all, while other (sometimes simpler GIFs) can artifact like crazy.  Any ideas on other troubleshooting steps I can try?

**Comment by shanecbauman** [comment]
Here are a few examples of the artifacts im getting:
![DeadlyMadeupBluejay-size_restricted](https://user-images.githubusercontent.com/41882311/93164672-fd349900-f6ce-11ea-89e7-ea7bf507ac7c.gif)
![InnocentEvergreenAmurratsnake-size_restricted](https://user-images.githubusercontent.com/41882311/93164681-032a7a00-f6cf-11ea-8142-5a02c9ee54db.gif)
![UnknownThriftyCottontail-size_restricted](https://user-images.githubusercontent.com/41882311/93164697-0aea1e80-f6cf-11ea-8b5a-5da618918568.gif)

**Comment by shanecbauman** [comment]
Ok I found the issue.  The GIFs I was having issues with were "optimized", which is a way to compress GIFs.  

A side effect to optimized GIFs is that some of the frames in the animation might not have the whole frame painted out.  In the case of the soccer ball in my last comment, some of the frames only had the parts of the image that moved from that last frame drawn out.  The rest of that frame would be transparent.

To get around this I used a tool called "Gifsicle" to optimize all my GIFs.  This essentially made sure every frame in my GIF was fully rendered to that specific point in the animation.  Every GIF I was having issues with is now running great.

Im not sure how big of a task it would be, but adding support for optimized GIFs would be super helpful.  Either way I hope this information helps someone down the line.
